Update package manager instructions with community-hosted repos (#42022)
kubernetes/website: content/en/docs/tasks/tools/install-kubectl-linux.md

The following methods exist for installing kubectl on Linux:

 - [Install kubectl binary with curl on Linux](#install-kubectl-binary-with-curl-on-linux)
 - [Install using native package management](#install-using-native-package-management)

>  - [Install using other package management](#install-using-other-package-management)

Reviewer: Why the leading whitespace? I'd leave it out.

   ```shell
   sudo apt-get update

>    sudo apt-get install -y apt-transport-https ca-certificates curl

Author: Added to other relevant documents as well.

   ```

2. Download the public signing key for the Kubernetes package repositories. The same signing key is used for all repositories so you can disregard the version in the URL:

   ```shell
   curl -fsSL https://pkgs.k8s.io/core:/stable:/{{< param "version" >}}/deb/Release.key | sudo gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
   ```

3. Add the appropriate Kubernetes `apt` repository. If you want to use Kubernetes version different than {{< param "version" >}},
   replace {{< param "version" >}} with the desired minor version in the command below:

   ```shell
   echo 'deb https://pkgs.k8s.io/core:/stable:/{{< param "version" >}}/deb/ /' | sudo tee /etc/apt/sources.list.d/kubernetes.list
   ```

   {{< note >}}

>    If you desire to upgrade kubectl to another minor release at some point, you'll need to change version in `/etc/apt/sources.list.d/kubernetes.list` before running `apt-get update` and `apt-get upgrade`. This procedure is described in [Changing the Kubernetes package repository](/docs/tasks/administer-cluster/kubeadm/change-package-repository/) document.

Author: I applied the suggestion that @puerco proposed.

{{< warning >}}
Kubernetes has two different package repositories starting from August 2023.
The Google-hosted repository is considered deprecated and we will stop
publishing packages to that repository starting with Kubernetes 1.30.0.
The community-hosted repositories are replacement for the Google-hosted
repository. There are some important considerations for the community-hosted
repositories:

- We strongly recommend using the community-hosted repositories especially for
  new setups. For existing setups, we recommend migrating to the community-hosted
  repositories. The migration process is described in the following document: TBD.
- The community-hosted repositories contain only packages starting with
  Kubernetes 1.24.0. Earlier versions are available only in the Google-hosted
  repository.
- We have a different repository for each Kubernetes minor release.
  Please take this into consideration when upgrading to the next minor release.

For more information about community-hosted repositories, we recommend
checking out the following document: TBD
{{< /warning >}}

{{< tabs name="kubectl_install" >}}
{{% tab name="Debian-based distributions" %}}

> ### Community-hosted repositories

Author: @sftim I'm not really sure I understand this comment, can you please elaborate?

1. Update the `apt` package index and install packages needed to use the Kubernetes `apt` repository:

   ```shell
   sudo apt-get update
   sudo apt-get install -y apt-transport-https ca-certificates curl
   ```

2. Download the public signing key for community-hosted repositories. The same signing key is used for all repositories so you can disregard the version in the URL:

   ```shell
   curl -fsSL https://pkgs.k8s.io/core:/stable:/{{< param "version" >}}/deb/Release.key | sudo gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
   ```

> 3. Add the appropriate community-hosted Kubernetes `apt` repository. If you want to use Kubernetes version different than {{< param "version" >}},

Author: I'm wondering if this is worth it. It is really about only changing one number in the given command which should be intuitive enough given that an example for the current version is provided. Looking from the user/reader perspective, switching to docs version, having to navigate again to the correct section, and reading it all over again is more complicated IMO.

{{% /tab %}}

{{% tab name="Red Hat-based distributions" %}}

1. Add the Kubernetes `yum` repository. If you want to use Kubernetes version different than {{< param "version" >}}, replace {{< param "version" >}} with the desired minor version in the command below:

```bash
cat <<EOF | sudo tee /etc/yum.repos.d/kubernetes.repo
[kubernetes]
name=Kubernetes
baseurl=https://pkgs.k8s.io/core:/stable:/{{< param "version" >}}/rpm/
enabled=1
gpgcheck=1
gpgkey=https://pkgs.k8s.io/core:/stable:/{{< param "version" >}}/rpm/repodata/repomd.xml.key
exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni
EOF
```

{{< note >}}

> If you desire to upgrade kubectl to another minor release at some point, you'll need to change version in `/etc/yum.repos.d/kubernetes.repo` before running `yum update`. This procedure is described in [Changing the Kubernetes package repository](/docs/tasks/administer-cluster/kubeadm/change-package-repository/) document.

Reviewer 2: ```suggestion
To upgrade kubectl to another minor release, you'll need to bump the version in `/etc/yum.repos.d/kubernetes.repo` before running `yum update`. This procedure is described in more detail in [its own document](/docs/tasks/administer-cluster/kubeadm/change-package-repository/).
```
